textacy/tm/topic_model.py

```diff
@@ -292,3 +300,135 @@ def topic_weights(self, doc_topic_matrix):
+
+        References:
+            .. Chuang, Jason, Christopher D. Manning, and Jeffrey Heer. "Termite:
+                Visualization techniques for assessing textual topic models."
+                Proceedings of the International Working Conference on Advanced
+                Visual Interfaces. ACM, 2012.
+            .. for sorting by "seriation", see https://arxiv.org/abs/1406.5370
+
+        .. seealso:: :func:`viz.termite_plot <textacy.viz.termite.termite_plot>`
+
+        TODO: `rank_terms_by` other metrics, e.g. topic salience or relevance
+        """
+        if highlight_topics is not None:
+            if isinstance(highlight_topics, int):
+                highlight_topics = (highlight_topics,)
+            elif len(highlight_topics) > 6:
+                raise ValueError('no more than 6 topics may be highlighted at once')
+        else:
+            highlight_topics = tuple()
+
+        # get topics indices
+        if topics == -1:
+            topic_inds = tuple(range(self.n_topics))
+        elif isinstance(topics, int):
+            topic_inds = (topics,)
+        else:
+            topic_inds = tuple(topics)
+
+        # get topic indices in sorted order
+        if sort_topics_by == 'index':
+            topic_inds = sorted(topic_inds)
+        elif sort_topics_by == 'weight':
+            topic_inds = tuple(topic_ind for topic_ind
+                               in np.argsort(self.topic_weights(self.transform(doc_term_matrix)))[::-1]
+                               if topic_ind in topic_inds)
+        else:
+            msg = 'invalid sort_topics_by value; must be in {}'.format(
+                {'index', 'weight'})
+            raise ValueError(msg)
+
+        # get column index of any topics to highlight in termite plot
+        highlight_cols = tuple(i for i in range(len(topic_inds))
+                               if topic_inds[i] in highlight_topics)
+
+        # get top term indices
+        if rank_terms_by == 'corpus_weight':
+            term_inds = np.argsort(np.ravel(doc_term_matrix.sum(axis=0)))[:-n_terms - 1:-1]
+        elif rank_terms_by == 'topic_weight':
+            term_inds = np.argsort(self.model.components_.sum(axis=0))[:-n_terms - 1:-1]
+        else:
+            msg = 'invalid rank_terms_by value; must be in {}'.format(
+                {'corpus_weight', 'topic_weight'})
+            raise ValueError(msg)
+
+        # get top term indices in sorted order
+        if sort_terms_by == 'weight':
+            pass
+        elif sort_terms_by == 'index':
+            term_inds = sorted(term_inds)
+        elif sort_terms_by == 'alphabetical':
+            term_inds = sorted(term_inds, key=lambda x: id2term[x])
+        elif sort_terms_by == 'seriation':
+            topic_term_weights_mat = np.array(
+                np.array([self.model.components_[topic_ind][term_inds]
+                          for topic_ind in topic_inds])).T
+            # calculate similarity matrix
```

Comment:
Yeah, that's my intention. However, since it is only used _here_ right now, I just kept it local. Fine by you to wait until there's the use case?